test/Amazon.Extensions.Configuration.SystemsManager.Integ/AppConfigEndToEndTests.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+using Amazon;
+using Amazon.AppConfig;
+using Amazon.AppConfig.Model;
+using Amazon.Extensions.NETCore.Setup;
+using Microsoft.Extensions.Configuration;
+using Xunit;
+
+namespace Amazon.Extensions.Configuration.SystemsManager.Integ
+{
+    public class AppConfigEndToEndTests
+    {
+        IAmazonAppConfig _appConfigClient = new AmazonAppConfigClient(RegionEndpoint.USWest2);
+
+        [Fact]
+        public async Task RefreshConfiguration()
+        {
+            var configSettings = new Dictionary<string, string>
+            {
+                { "key1", "value1" },
+                { "key2", "value2" }
+            };
+            (string applicationId, string environmentId, string configProfileId) = await CreateAppConfigResourcesAsync("RefreshWebAppTest", configSettings);
+            try
+            {
+                var builder = new ConfigurationBuilder()
+                                .AddAppConfig(applicationId, environmentId, configProfileId, new AWSOptions {Region = RegionEndpoint.USWest2 }, TimeSpan.FromSeconds(5));
+
+                var configuration = builder.Build();
+
+                Assert.Equal("value1", configuration["key1"]);
+
+                configSettings["key1"] = "newValue1";
+                var versionNumber = await CreateNewHostedConfig(applicationId, configProfileId, configSettings);
+                await PerformDeploymentAsync(applicationId, environmentId, configProfileId, versionNumber);
+
+                // Wait for ConfigProvider to perform the reload
+                await Task.Delay(10);
```

Comment:
nit: for readability
```suggestion
                await Task.Delay(TimeSpan.FromMilliseconds(5));

```

Reply:
Funny that you are right about the readability. It was actually supposed to be seconds since ConfigurationProvider reloads every 5 seconds. Adjusted to 7 seconds.

Reply:
:O 

[Task.Delay](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.delay?view=net-6.0#system-threading-tasks-task-delay(system-int32)) actually takes milliseconds, glad we caught this. Yet another reason of using `TimeSpan`.